### DISCUSSION TITLE
Uploaded file without extension takes .txt estension by defatult

### DISCUSSION THREAD

**Discussion by GhaziTriki** [pocketbase/pocketbase#5991] [Q&A]
Hello,

When we upload a file without an extension it takes .txt extension by default. While the user might keep the file extension-less for a purpose, why that txt extension is added? Isn't it better to keep it without that extension?

**Comment by ganigeorgiev** [comment]
The extension is normalized in case it is missing or contains invalid characters and in this case it is determined based on the file content, if possible (in a similar manner to the MIME type checks). While technically the extension doesn't really matter, it is helpful to always have an extension since the UI preview relies on it. There are no plans to change the current behavior unless it is causing an actual issue.

**Reply by GhaziTriki** [reply]
In our use case we upload shell files without an extension to be stored in `/etc/cron.d`. The txt extension is not annoying as final files are always renamed, but strange.
